In [1]:
import pandas as pd

df = pd.read_csv("datasets/olist_products_dataset.csv")

df


product_id              product_category_name  \
0      1e9e8ef04dbcff4541ed26657ea517e5                         perfumaria   
1      3aa071139cb16b67ca9e5dea641aaa2f                              artes   
2      96bd76ec8810374ed1b65e291975717f                      esporte_lazer   
3      cef67bcfe19066a932b7673e239eb23d                              bebes   
4      9dc1a7de274444849c219cff195d0b71              utilidades_domesticas   
...                                 ...                                ...   
32946  a0b7d5a992ccda646f2d34e418fff5a0                   moveis_decoracao   
32947  bf4538d88321d0fd4412a93c974510e6  construcao_ferramentas_iluminacao   
32948  9a7c6041fa9592d9d9ef6cfe62a71f8c                    cama_mesa_banho   
32949  83808703fc0706a22e264b9d75f04a2e             informatica_acessorios   
32950  106392145fca363410d287a815be6de4                    cama_mesa_banho   

       product_name_lenght  product_description_lenght  product_photos_qty  \
0                     40.0                       287.0                 1.0   
1                     44.0                       276.0                 1.0   
2                     46.0                       250.0                 1.0   
3                     27.0                       261.0                 1.0   
4                     37.0                       402.0                 4.0   
...                    ...                         ...                 ...   
32946                 45.0                        67.0                 2.0   
32947                 41.0                       971.0                 1.0   
32948                 50.0                       799.0                 1.0   
32949                 60.0                       156.0                 2.0   
32950                 58.0                       309.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
0                 225.0               16.0               10.0   
1                1000.0               30.0               18.0   
2                 154.0               18.0                9.0   
3                 371.0               26.0                4.0   
4                 625.0               20.0               17.0   
...                 ...                ...                ...   
32946           12300.0               40.0               40.0   
32947            1700.0               16.0               19.0   
32948            1400.0               27.0                7.0   
32949             700.0               31.0               13.0   
32950            2083.0               12.0                2.0   

       product_width_cm  
0                  14.0  
1                  20.0  
2                  15.0  
3                  26.0  
4                  13.0  
...                 ...  
32946              40.0  
32947              16.0  
32948              27.0  
32949              20.0  
32950               7.0  

[32951 rows x 9 columns]

Há dados faltantes neste dadaset?

In [2]:

df.isnull().any().any()

True

Quantos dados faltantes temos no dataset?

In [3]:
df.isnull().sum().sum()

2448

Retirar os dados faltantes

In [4]:
df = df.dropna()

In [5]:
categorias = df['product_category_name']
categorias



0                               perfumaria
1                                    artes
2                            esporte_lazer
3                                    bebes
4                    utilidades_domesticas
                       ...                
32946                     moveis_decoracao
32947    construcao_ferramentas_iluminacao
32948                      cama_mesa_banho
32949               informatica_acessorios
32950                      cama_mesa_banho
Name: product_category_name, Length: 32340, dtype: object

Temos 32951 dados no dataset em diversas caterigorias que podem ser as mesmas ou nao

In [6]:
#  Qual é a categoria que mais vende?
categorias.describe()


count               32340
unique                 73
top       cama_mesa_banho
freq                 3029
Name: product_category_name, dtype: object

In [7]:
categorias.value_counts()

product_category_name
cama_mesa_banho                  3029
esporte_lazer                    2867
moveis_decoracao                 2657
beleza_saude                     2444
utilidades_domesticas            2335
                                 ... 
fashion_roupa_infanto_juvenil       5
casa_conforto_2                     5
pc_gamer                            3
seguros_e_servicos                  2
cds_dvds_musicais                   1
Name: count, Length: 73, dtype: int64

In [8]:
df['volume_produto'] = df['product_length_cm'] * df['product_height_cm'] * df['product_width_cm']

C:\Users\Drac3\AppData\Local\Temp\ipykernel_14608\3362806198.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['volume_produto'] = df['product_length_cm'] * df['product_height_cm'] * df['product_width_cm']


In [9]:
df

product_id              product_category_name  \
0      1e9e8ef04dbcff4541ed26657ea517e5                         perfumaria   
1      3aa071139cb16b67ca9e5dea641aaa2f                              artes   
2      96bd76ec8810374ed1b65e291975717f                      esporte_lazer   
3      cef67bcfe19066a932b7673e239eb23d                              bebes   
4      9dc1a7de274444849c219cff195d0b71              utilidades_domesticas   
...                                 ...                                ...   
32946  a0b7d5a992ccda646f2d34e418fff5a0                   moveis_decoracao   
32947  bf4538d88321d0fd4412a93c974510e6  construcao_ferramentas_iluminacao   
32948  9a7c6041fa9592d9d9ef6cfe62a71f8c                    cama_mesa_banho   
32949  83808703fc0706a22e264b9d75f04a2e             informatica_acessorios   
32950  106392145fca363410d287a815be6de4                    cama_mesa_banho   

       product_name_lenght  product_description_lenght  product_photos_qty  \
0                     40.0                       287.0                 1.0   
1                     44.0                       276.0                 1.0   
2                     46.0                       250.0                 1.0   
3                     27.0                       261.0                 1.0   
4                     37.0                       402.0                 4.0   
...                    ...                         ...                 ...   
32946                 45.0                        67.0                 2.0   
32947                 41.0                       971.0                 1.0   
32948                 50.0                       799.0                 1.0   
32949                 60.0                       156.0                 2.0   
32950                 58.0                       309.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
0                 225.0               16.0               10.0   
1                1000.0               30.0               18.0   
2                 154.0               18.0                9.0   
3                 371.0               26.0                4.0   
4                 625.0               20.0               17.0   
...                 ...                ...                ...   
32946           12300.0               40.0               40.0   
32947            1700.0               16.0               19.0   
32948            1400.0               27.0                7.0   
32949             700.0               31.0               13.0   
32950            2083.0               12.0                2.0   

       product_width_cm  volume_produto  
0                  14.0          2240.0  
1                  20.0         10800.0  
2                  15.0          2430.0  
3                  26.0          2704.0  
4                  13.0          4420.0  
...                 ...             ...  
32946              40.0         64000.0  
32947              16.0          4864.0  
32948              27.0          5103.0  
32949              20.0          8060.0  
32950               7.0           168.0  

[32340 rows x 10 columns]

In [13]:
#maiores produtos

df_sorted = df.sort_values(by='volume_produto', ascending=False)
df_sorted.head(50)

product_id  \
12802  256a9c364b75753b97bee410c9491ad8   
25524  c1e0531cb1864fd3a0cae57dca55ca80   
17303  3eb14e65e4208c6d94b7a32e41add538   
10697  0b48eade13cfad433122f23739a66898   
12146  f227e2d44f10f7dad30fb4dfa839e7a2   
32532  90c1b4e040d1d1c45897ec2dad4a809d   
16728  99ff40856c47a638df807c0a144470cc   
3770   c6fdec160d0f8f488d9041316c85051d   
7234   8d6f2c3454002d3f5aa7479a7fad7794   
28145  0e9dfb804bafa3d68ef3ee7a621abfb2   
27266  ab495f166205a883ffe5ab0b5b55f867   
20574  a3cd9517ebf5a50dca25acce54f3b171   
23988  ab50aa8d9af0addb7f96149c8b8cee86   
1159   f97ad9066c718a6cef93dfcf253d3e0d   
25978  f18f40e6344e72cfd537f21d9745e8cc   
24685  c8d4b2652543be14a917376a17f426c2   
30958  04abe952a7292c9914a0ddade6371114   
9356   671be61e4013fc77e2d3586ffc1e51f1   
17314  2a511952313989fa097fb038cd758dc5   
23942  f3f12eabcd090650146791b249f48880   
6212   e0c47a145f4748e4d7a418502d65aa35   
13987  d13816a79f03bf978283d80fbeae06f8   
6012   d5536c1422144c4fc6fe001224b21ba8   
29085  107fde0930956120d1e13dd1062fbb46   
30518  013b5dd7d319685753e2c12292975615   
1514   dcfeedf441c38e5e7e58ffce194af2bb   
217    6e9c52b407fea24064f7ab2b144fbf8c   
26970  b17808303e15dd50538c011b44295427   
10248  3d77287739b6bf1ac163ce2d77570ada   
6728   5caaa02918381c4ad59d38fdcd4a28b9   
15581  b7a64f15c532c9eacbaa11cbef1bc969   
13324  27761f21b03a32be1aad6ebaebddd747   
24886  5dcd17b12ed2175d4fe6157e047d7be2   
10463  0d5837a2580f92e18e2d1b5f7fa70c2d   
6004   368645a47a15770520ad12d9ea77c2f2   
10708  b6e406be1aa00db8be5dd3ec6b524d4d   
7535   9dfef86fb34051388a7263a31642386c   
8218   9fa7e05ae5cddeb74ff2fe5046b742db   
2537   0ed9a84687fad2d921b09b0ebcc4cded   
21443  a9b7d6445491f7108ce079d4f783dbf1   
7422   bbc5419bc92d76e27795c9ec2c233c64   
20222  48c9b5b0aff66a99d74fbc6e2c600b33   
12376  773d43a287d64831b05013a73a33f229   
4814   0984eaa8480e41aded35bd7b5131a1c1   
26495  d15010eecc80e3a843ccb15865c328da   
21484  67f1a036affe9a75b73dfddf0386893a   
27326  46e24ce614899e36617e37ea1e4aa6ff   
24029  1d16c0509e7011656e03c6a582ff6667   
27698  97c948ebc8c04b26b7bbb095d4228f2a   
29578  dc861f18267cce21f3b70f9f7990b8ce   

                                product_category_name  product_name_lenght  \
12802                           utilidades_domesticas                 59.0   
25524                                     moveis_sala                 50.0   
17303                                     moveis_sala                 51.0   
10697                                moveis_decoracao                 28.0   
12146                                     moveis_sala                 51.0   
32532                                     moveis_sala                 57.0   
16728                                     moveis_sala                 54.0   
3770                                      moveis_sala                 57.0   
7234                                      moveis_sala                 52.0   
28145                                     moveis_sala                 17.0   
27266                                eletrodomesticos                 57.0   
20574                           utilidades_domesticas                 52.0   
23988                                      cool_stuff                 34.0   
1159                                 moveis_decoracao                 63.0   
25978                                moveis_decoracao                 63.0   
24685                                  consoles_games                 41.0   
30958                                        pet_shop                 33.0   
9356                                    esporte_lazer                 28.0   
17314                                    beleza_saude                 51.0   
23942                                   esporte_lazer                 47.0   
6212                                       cool_stuff                 56.0   
13987                                     moveis_sala                 45.0   
6012                                 move

In [12]:
df['volume_produto'].sort_values(ascending=False)

12802    296208.0
25524    294000.0
17303    294000.0
10697    294000.0
12146    294000.0
           ...   
4415        352.0
20028       352.0
20041       352.0
961         288.0
32950       168.0
Name: volume_produto, Length: 32340, dtype: float64

In [15]:
#produtos mais pesados
df_sorted = df.sort_values(by='product_weight_g', ascending=False)
df_sorted.head(50)


product_id  \
25166  26644690fde745fc4654719c3904e1db   
24268  cb26d15d1b6eabaac7c0803774245884   
1701   1c57458e824ca3d974ec1831a1a55e72   
27266  ab495f166205a883ffe5ab0b5b55f867   
19619  cdb8d3c880b6639a70764c6734e6bb69   
1714   c04e948c6900ce99ac47d89b3b6d70cd   
9363   8f73613d06e3a557da0249015cbae6b6   
12376  773d43a287d64831b05013a73a33f229   
32262  ebf1c13032246ea801765e8cb5417365   
27230  e5f2d52b802189ee658865ca93d83a8f   
9373   038e3a669bb495e56d9d609d874fb668   
1738   0a859d8dc68f6a746b4709217110c439   
19753  31d22f91c17551712034a105e70ab838   
11327  8d4e92265a16e69a1e1d76e67e46d72f   
7535   9dfef86fb34051388a7263a31642386c   
19834  0f3777229e40207d248acfd6e1f716b6   
8218   9fa7e05ae5cddeb74ff2fe5046b742db   
8644   59f09deeea7c839a9212a19f3416b892   
19875  9f82fdef2a36bb15d22d70af1aedb3ce   
32179  d7f8b7443f6ae5796dabe493521a1e79   
5465   df872c596e00cd0160e6c8eb5424739f   
26912  bad9cd5ad615c0b5ba87448e03ec954c   
30340  660422061e06da17ca6101e9d6b7aae8   
14992  07f7c5fe95aa4a3b8ea56a5119546939   
20222  48c9b5b0aff66a99d74fbc6e2c600b33   
9356   671be61e4013fc77e2d3586ffc1e51f1   
2417   343c15a347e523f2b6cf38a5db81e179   
1514   dcfeedf441c38e5e7e58ffce194af2bb   
27326  46e24ce614899e36617e37ea1e4aa6ff   
27858  0c5aecba8eab00eb055c6e2e7b9b4346   
16575  ab14f3763110d218ea7862647bd450af   
32532  90c1b4e040d1d1c45897ec2dad4a809d   
18791  fe5915b903037937488e868f2dde0ff3   
27698  97c948ebc8c04b26b7bbb095d4228f2a   
32491  2b752ed328ea866e4721ca4e236a416c   
29578  dc861f18267cce21f3b70f9f7990b8ce   
24861  961930544d39899c18cfdbc336aeb15e   
16450  e6506b15fb88b22adb8f9dcdb20e532a   
23581  a8baceb529f7e2a5c770cc5b4e3da35d   
16365  f18e0f3757837733f36da5143ddccd73   
12105  c75c215de0f9ab2c103fc57fb82f47ad   
29682  17383e180cbea9ef103d90a72754cd54   
19023  bc75ffbc500b02276f9c55aef39014e9   
12146  f227e2d44f10f7dad30fb4dfa839e7a2   
16206  df4566130f96bba07730095eef42389e   
7422   bbc5419bc92d76e27795c9ec2c233c64   
19233  1ab21d1c80e214bc2cedfcb7a5ade550   
27449  7d96d1abbde52bb350c688d0d049df07   
4185   c3eb7064c36408a9bfc77a42d6ab993c   
15982  c100e5fef1abb5e1c5054d1dac2d83ac   

                                product_category_name  product_name_lenght  \
25166                                 cama_mesa_banho                 59.0   
24268                                   esporte_lazer                 54.0   
1701                                         pet_shop                 57.0   
27266                                eletrodomesticos                 57.0   
19619                                    beleza_saude                 25.0   
1714                                       cool_stuff                 33.0   
9363                                moveis_escritorio                 54.0   
12376                                    beleza_saude                 55.0   
32262                                 casa_construcao                 39.0   
27230                                        pet_shop                 56.0   
9373                                    esporte_lazer                 49.0   
1738                                    esporte_lazer                 50.0   
19753                                moveis_decoracao                 29.0   
11327                           instrumentos_musicais                 45.0   
7535                                     beleza_saude                 57.0   
19834                                   esporte_lazer                 64.0   
8218                                     beleza_saude                 55.0   
8644                            utilidades_domesticas                 58.0   
19875                              ferramentas_jardim                 57.0   
32179                   industria_comercio_e_negocios                 56.0   
5465   moveis_cozinha_area_de_servico_jantar_e_jardim                 39.0   
26912                                    beleza_saude                 59.0   
30340                                   e

In [16]:
df['product_weight_g'].sort_values(ascending=False)

25166    40425.0
24268    30000.0
1701     30000.0
27266    30000.0
19619    30000.0
          ...   
15675        2.0
14997        0.0
9769         0.0
32079        0.0
13683        0.0
Name: product_weight_g, Length: 32340, dtype: float64

In [17]:
df['product_name_lenght'].describe()

count    32340.000000
mean        48.476592
std         10.245699
min          5.000000
25%         42.000000
50%         51.000000
75%         57.000000
max         76.000000
Name: product_name_lenght, dtype: float64

In [18]:
df['product_description_lenght'].describe()

count    32340.000000
mean       771.492393
std        635.124831
min          4.000000
25%        339.000000
50%        595.000000
75%        972.000000
max       3992.000000
Name: product_description_lenght, dtype: float64

In [24]:
max_product_description_lenght = df['product_description_lenght'].max()
df_product_description_lenght = df.loc[df['product_description_lenght'] == max_product_description_lenght]
df_product_description_lenght

product_id product_category_name  \
31404  47d52bb24ef8a3aa09724f00604be3ba          dvds_blu_ray   

       product_name_lenght  product_description_lenght  product_photos_qty  \
31404                 30.0                      3992.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
31404            2600.0               41.0               11.0   

       product_width_cm  volume_produto  
31404              32.0         14432.0

In [19]:
df['product_photos_qty'].describe()

count    32340.000000
mean         2.188961
std          1.736787
min          1.000000
25%          1.000000
50%          1.000000
75%          3.000000
max         20.000000
Name: product_photos_qty, dtype: float64

In [22]:
max_photos_qty = df['product_photos_qty'].max()
df_max_photos = df.loc[df['product_photos_qty'] == max_photos_qty]
df_max_photos

product_id product_category_name  \
9022  f95d5d21561ea085ba1e1a4e53840844            brinquedos   

      product_name_lenght  product_description_lenght  product_photos_qty  \
9022                 62.0                       679.0                20.0   

      product_weight_g  product_length_cm  product_height_cm  \
9022            8900.0               32.0               49.0   

      product_width_cm  volume_produto  
9022              34.0         53312.0

In [ ]:
import pandas as pd

df_items = pd.read_csv("datasets/olist_order_items_dataset.csv")

df_items